### ISSUE TITLE
Cannot call decoder.predict in testing phase

### ISSUE THREAD

**Issue by pachpandepriti** [keras-team/keras#12415]
![image](https://user-images.githubusercontent.com/29904255/53972954-d1922500-40cd-11e9-8184-2895d7607a45.png)
 training works great decoder.summary also looks correct

**Comment by disign-ming** [comment]
encode.predict Used only with the training model, if you train your own classifier, you can use the following code to find the predicted value and the corresponding label.
test_gen2 = datagen_test.flow_from_directory(directory=test_dir2,
                                                  target_size=(224,224),
                                                  batch_size=1,
                                                  class_mode='categorical',
                                                  shuffle=False)
test_gen2.reset()
pred = new_model.predict_generator(test_gen2, verbose=1)

predicted_class_indices = np.argmax(pred, axis=1)
labels = (generator_train.class_indices)
label = dict((v,k) for k,v in labels.items())

 
predictions = [label[i] for i in predicted_class_indices]
filenames = test_gen2.filenames
for idx in range(len(filenames )):
    print('predict  %d' % (int(predictions[idx])))
    print('title    %s' % filenames[idx])
    print('')
